src/ln/peer_handler.rs

```diff
@@ -786,6 +789,19 @@ impl<Descriptor: SocketDescriptor> PeerManager<Descriptor> {
 						for msg in update_fail_malformed_htlcs {
 							peer.pending_outbound_buffer.push_back(peer.channel_encryptor.encrypt_message(&encode_msg!(msg, 135)));
 						}
+						debug_assert!(update_fee.is_none());
+						peer.pending_outbound_buffer.push_back(peer.channel_encryptor.encrypt_message(&encode_msg!(commitment_signed, 132)));
+						Self::do_attempt_write_data(&mut descriptor, peer);
+						continue;
+					},
+					Event::SendUpdateFee { ref node_id, ref msg, ref commitment_signed } => {
```

Comment:
Why add a new event for SendUpdateFee when you've added it to CommitmentUpdate. Why not just generate a UpdateHTLCs event?